### ISSUE TITLE
report an error when a type variable cannot be resolved on a call-site

### ISSUE THREAD

**Issue by KotlinIsland** [astral-sh/ty#1230]
### Summary

```py
class A[T]:
    def __init__(self, *t: T): ...

A()  # no error, expect: type variable `T@A` could not be resolved, please provide it explicitly
```

currently it becomes `Unknown` without an indication

**Comment by AlexWaygood** [comment]
I agree we should do this.

We can't do it _right_ now because our current constraints solver is quite naïve, so there are many places where we should be solving a TypeVar but currently can't. That means that if we implemented this check now, it would lead to way too many false-positive diagnostics on user code. We should reconsider once the new constraints solver is further along (#623)
